core/templates/dev/head/exploration_editor/exploration_editor.html

```diff
@@ -424,6 +424,43 @@
       </div>
     </script>
 
+    <script type="text/ng-template" id="modals/saveValidationFail">
+        <div class="modal-header">
+            <h3>Saving error - non-strict validation failure</h3>
+        </div>
+
+        <div class="modal-body">
+            <p>
+                Sorry, an error has occurred. Please refresh the page to get back to the most recent ‘good’ state.
+            </p>
+        </div>
+
+        <div class="modal-footer">
+            <button class="btn btn-default" ng-click="close()">Close</button>
+        </div>
+    </script>
+
+    <script type="text/ng-template" id="modals/saveVersionMismatch">
+        <div class="modal-header">
+            <h3>Saving error - Version mismatch</h3>
+        </div>
+
+        <div class="modal-body">
+            <p>
+                Sorry someone else has saved a new version of this exploration and so your pending changes cannot be saved. The lost changes are displayed below:
+            </p>
+
+            <div class="lost-changes" ng-bind-html="lostChangesHtml">
```

Comment:
I would make a suggestion that the user can copy and paste the lost changes before discarding them.